src/containers/App.jsx

```diff
@@ -31,8 +34,8 @@ const App = ({ dispatch, links, addLink, removeLink }) => (
 
 const mapStateToProps = state => (
   {
-    // I can tell you more about this if you want, but use whats called selectors when ever you grab something off of state
-    links: state.linkReducer.links
+    // Call get Links and pass it a ref to state and let it do its thing. You now have links from state
+    links: getLinks(state)
```

Comment:
The benefit here is that you now done have to know how your state is shaped. You can know that when you call getLinks you will get the links... or if your state shape changes down the road. you dont have to go into every single file that uses links and change it to the new shape. The selector does that for you